spinn_front_end_common/utility_models/reverse_ip_tag_multi_cast_source.py

```diff
@@ -146,6 +148,17 @@ def __init__(
         # Keep the vertices for resuming runs
         self._machine_vertices = list()
 
+    def _validate_send_buffer_times(self, n_keys, send_buffer_times):
+        if send_buffer_times is None:
+            return
+        if len(send_buffer_times) and hasattr(send_buffer_times[0], "__len__"):
+            if len(send_buffer_times) != n_keys:
```

Comment:
why use n keys instead of self._n_atoms. i appreciate they're the same value. but n keys is dubious. if it had multiple outgoing partitions, then n keys wouldn't be correct.

Reply:
Ok changed which also reduces a param for the method.